### ISSUE TITLE
Fixing Potential Issue with PATH Generation

### ISSUE THREAD

**Issue by sonic2kk** [flathub/com.valvesoftware.Steam.Utility.steamtinkerlaunch#137]
Hello!

A user recently reported in sonic2kk/steamtinkerlaunch#1005 that STL was unable to find MangoHud, I believe it is Flatpak MangoHud as it was at the path `/usr/lib/extensions/vulkan/MangoHud/bin/mangohud`. This can be worked around by manually setting the path to the MangoHud binary in the config file (this user could not get the file browser dialog to appear for them for some reason, so they had to input it in the config file manually).

This user was able to narrow down a potential cause of the issue being some incorrect path generation. On their system, the path ends up with a trailing colon somewhere along the way, and STL sets `PATH` to `/app/bin:/app/utils/bin:/usr/bin:/usr/lib/extensions/vulkan/MangoHud/bin/=/app/bin:/app/utils/bin:/usr/bin:/usr/lib/extensions/vulkan/MangoHud/bin/:/usr/lib/extensions/vulkan/MangoHud/bin:/usr/lib/extensions/vulkan/gamescope/bin::/app/utils/bin/` -- Note the `::` at the end of the path. This comes from `emptyVars` in a log snippet they attached (https://github.com/sonic2kk/steamtinkerlaunch/issues/1005#issuecomment-1879745588)

It seems this stray colon comes from the following logic in `emptyVars` ([link](https://github.com/sonic2kk/steamtinkerlaunch/blob/7d63df72bc71fb8ab956ed6f32ce45b5e61dc773/steamtinkerlaunch#L24417-L24422) because GitHub won't embed the code snippet):

```bash
FPPATH="/app/utils/bin/"
if [ -d "$FPPATH" ]; then
    PATH="$STLPATH:$FPPATH"
else
    PATH="$STLPATH"
fi
```

It appears that `STLPATH` is the culprit for the trailing colon. Note how `FPPATH` here matches the end of their log, so it would make sense that `STLPATH` has a trailing colon.

We should be able to solve this by using parameter expansion to remove the trailing colon, like so: `${STLPATH%:}`. This will only remove a colon from the end of the string if it is present, so strings like `/path/one:/path/two:/path/three` won't lose any colons and will remain intact, but paths like `/normal/path:/trailing/colon:` will have the colon correctly stripped from the end.

<hr>

I am opening this issue describing the potential issue and potential fix for two reasons:
1. I emphasise potential issue, with no disrespect to the OP of the issue, simply because this logic has been in place for a long time without any reported issues. Therefore I am a bit cautious of changing it in case it doesn't need to be changed, or in case any other parts of the code rely on this seemingly faulty logic.
2. I don't have a machine with Steam Flatpak to test, so I wanted to run this issue and potential fix by you, with a documented issue opened on this repository, and get your input. I would also feel more confident if you could test this fix and if it is valid, it can get merged into upstream STL. I did ask OP if they wanted to make a PR over at sonic2kk/steamtinkerlaunch as they were the one that identified the issue.

Flatpaks aren't my field at all so any input you can give here is really appreciated (and seriously, thank you for all the work you do to help maintain STL Flatpak  :smile:)

**Comment by HanPrower** [comment]
Whilst I don't doubt the user may have had issues with MangoHUD if I backup and start fresh with STL, restart Steam, and then run STL it finds MangoHUD fine with the current version (v14.0.20231209-1). Don't need to locate it, it just populates correctly.

I'd expect lots of complaints if the PATH wasn't working correctly for things like mangohud and gamescope, etc. but sure, if you want to make the change I'll update the Flatpak and we'll see if it makes a difference for them. Might just be a weird edge-case.

**Comment by sonic2kk** [comment]
> Don't need to locate it, it just populates correctly. I'd expect lots of complaints if the PATH wasn't working correctly for things like mangohud and gamescope, etc.

I agree, I figured it would've come up before now. But since you're open to it (i.e. it doesn't set off any immediate alarm bells) and it should be a fairly straightforward, isolated change, I think it's safe to move forward with it.

I'll wait a few days and see if they have time to open a PR, otherwise I'll make the change myself. Once the change is done I'll ping here and close this issue.

Thanks! :smile:

**Comment by sonic2kk** [comment]
Change to remove trailing colon from `STLPATH` as described earlier was made in https://github.com/sonic2kk/steamtinkerlaunch/commit/07edefb8a70b015011474de858aead3bb5080726 (`v14.0.20230110-1`).

Hopefully this resolves the issue for the user, and I agree with you that this is probably an edge case. Given that it worked in your testing without any tweaks, I hope this change is enough to resolve the problem. Otherwise manually setting the MangoHud binary may have to do as a workaround for these edge cases (STL allows you to manually set it for this reason, in case it can't find it :slightly_smiling_face:)
